File: java/hamming/src/main/java/Hamming.java
class Hamming {

    private String leftStrand;
    private String rightStrand;

    Hamming(String leftStrand, String rightStrand) {
        this.leftStrand = leftStrand;
        this.rightStrand = rightStrand;
        if(rightStrand.length() ==0 && leftStrand.length() !=0 ){
            throw new IllegalArgumentException("right strand must not be empty.");
        }
        if(leftStrand.length() ==0 && rightStrand.length() !=0 ){
            throw new IllegalArgumentException("left strand must not be empty.");
        }
        if(rightStrand.length() != leftStrand.length()){
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }
    }

    int getHammingDistance() {
        int distance = 0;
        for(int i=0;i<leftStrand.length();i++){
          if(leftStrand.charAt(i) != rightStrand.charAt(i)){
            distance++;
          }
        }
        return distance;
    }
}

File: javascript/hamming/hamming.js
export const compute = (a, b) => {
  let c = 0;
  if (a === b) {
    return c;
  } else if (a.length !== b.length) {
    if (a.length == 0) {
      throw new Error("left strand must not be empty");
    } else if (b.length == 0) {
      throw new Error("right strand must not be empty");
    } else {
      throw new Error("left and right strands must be of equal length");
    }
  } else {
    return a.split("").filter((item, index) => item != b.charAt(index)).length;
  }
};